src/server/editorServices.ts

```diff
@@ -693,18 +693,24 @@ namespace ts.server {
         }
 
         getDefaultProjectForFile(fileName: NormalizedPath, ensureProject: boolean): Project | undefined {
-            let scriptInfo = this.getScriptInfoForNormalizedPath(fileName);
-            if (ensureProject && (!scriptInfo || scriptInfo.isOrphan())) {
-                this.ensureProjectStructuresUptoDate();
-                scriptInfo = this.getScriptInfoForNormalizedPath(fileName);
-                if (!scriptInfo) {
-                    return Errors.ThrowNoProject();
-                }
-                return scriptInfo.getDefaultProject();
-            }
+            return ensureProject ? this.ensureDefaultProjectForFile(fileName) : this.tryGetDefaultProjectForFile(fileName);
+        }
+
+        tryGetDefaultProjectForFile(fileName: NormalizedPath): Project | undefined {

```

Comment:
i would mark these as `/* @internal */`. both `tryGetDefaultProjectForFile` and `ensureDefaultProjectForFile`